src/components/graph/menus/network-modification-node-editor.js

```diff
@@ -353,6 +356,18 @@ const NetworkModificationNodeEditor = () => {
             });
     }, [currentTreeNode, snackError]);
 
+    useEffect(() => {
+        if (currentTreeNode !== undefined) {
+            fetchVoltageLevels(studyUuid, currentTreeNode.id).then((array) => {
+                setVoltageLevelOptions(
+                    array.sort((a, b) => a.id.localeCompare(b.id))
+                );
+            });
+        } else {
+            setVoltageLevelOptions([]);
+        }
+    }, [studyUuid, currentTreeNode]);
```

Comment:
We don't need to fetch VLs for form dialogs components when navigating between nodes I think. `withVLs` is called when needed, i.e when the concerned dialog is openned.
I adapted the code of `withEquipmentModificationOptions`, which works like a charm, for `withSubstations` and `withLines` I think it's better to be homogenous here.

Reply:
fixed